wolfcrypt/src/asn.c

```diff
@@ -3488,20 +3489,42 @@ int wc_RsaPublicKeyDecode(const byte* input, word32* inOutIdx, RsaKey* key,
     }
 #endif /* OPENSSL_EXTRA */
 
-    if (GetInt(&key->n,  input, inOutIdx, inSz) < 0)
-        return ASN_RSA_KEY_E;
-    if (GetInt(&key->e,  input, inOutIdx, inSz) < 0) {
-        mp_clear(&key->n);
-        return ASN_RSA_KEY_E;
+    if (key) {
+        if (GetInt(&key->n, input, inOutIdx, inSz) < 0)
+            return ASN_RSA_KEY_E;
+        if (GetInt(&key->e, input, inOutIdx, inSz) < 0) {
+            mp_clear(&key->n);
+            return ASN_RSA_KEY_E;
+        }
+    }
+    else {
```

Comment:
Why are we allowing a null RSA key and still parsing input?  That seems like a programming error.

Reply:
I've reverted the changes to `wc_RsaPublicKeyDecode` and instead added a new API for getting the public key size using a new API `wc_RsaPublicKeyDecodeSize`. The goal is to parse out the public key modulus size and not use any big integer math.